BatchAccountIdentity constructor breaking change

paterasMSFT opened the pull request:
The previous release introduced a breaking type ambiguity to BatchAccountIdentity's constructor. This PR corrects that.

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [x] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [x] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

paterasMSFT committed:
Fixed BatchAccountIdentity constructor breaking change.

Changed test location for CloudService test.

Update changelog and version.

Added more legacy constructors.

Fixed constructor and test

paterasMSFT committed:
Updated session records

Session records

dpwatrous committed:
Add another 2 argument constructor for BatchAccountIdentity

dpwatrous committed:
Mark legacy identity class and related constructor as obsolete

Added nuget config to fix package build order issue.

Revert "Added nuget config to fix package build order issue."

This reverts commit b59ade3181bf2f3cce37462a2f456200f2038d83.

Updated batch version

Revert "Updated batch version"

This reverts commit 86a050d1c3d18fc673615b2ebd6ec1dcadf9e917.

dpwatrous committed:
Mark legacy identity class and related constructor as obsolete

paterasMSFT committed:
Added nuget config to fix package build order issue.

paterasMSFT committed:
Revert "Added nuget config to fix package build order issue."

This reverts commit b59ade3181bf2f3cce37462a2f456200f2038d83.

paterasMSFT committed:
Updated batch version

paterasMSFT committed:
Updated package version

paterasMSFT committed:
Merge branch 'mipatera/batchaccountidentityfix' of https://github.com/Azure/azure-sdk-for-net into mipatera/batchaccountidentityfix

paterasMSFT committed:
Revert "Merge branch 'mipatera/batchaccountidentityfix' of https://github.com/Azure/azure-sdk-for-net into mipatera/batchaccountidentityfix"

This reverts commit fcbcebba5c12a98083e1d678037df8b31815a912, reversing
changes made to 0fbd7e667b93300cd8c56442f4bb888666206aa4.

wiboris reviewed (approved):
The changes look good

paterasMSFT committed:
Removed GeneratePackageOnBuild

Revert "Set RestorePackagePath"

This reverts commit f69ea24ed26d600c35d2df438804a8e7f121b731.

Removed GeneratePackageOnBuild

This resolved a packing issue. See [here](https://github.com/dotnet/sdk/issues/10335) for more information.

paterasMSFT committed:
Updated session recordings.

Updated location filter value to fix scenario test.

paterasMSFT committed:
Updated version

paterasMSFT commented:
Creating [new PR from a new branch](https://github.com/Azure/azure-sdk-for-net/compare/mipatera/newbatchaccountidentityfix?expand=1) to clean up commit history which involved merges.
